docs/deploy-guide/argocd.md

```diff
@@ -0,0 +1,169 @@
+
+If you already have [ArgoCD][argocd] deployed and available to be used then you
+can skip this section.
+
+The following command will do an initial deployment of ArgoCD that can
+then be customized further.
+
+```bash title="installing ArgoCD"
+kubectl kustomize --enable-helm https://github.com/rackerlabs/understack/bootstrap/argocd/ | kubectl apply -f -
+```
+
+## Configuring your Global and/or Site Cluster in ArgoCD
+
+For [ArgoCD][argocd] to be able to deploy to your cluster, you must define your cluster in
+ArgoCD. You can do this one of two ways, via the `argocd` CLI tool or via the
+[Declarative Setup][argocd-decl-setup]. We are embracing GitOps so the declarative
+is what we'll use.
+
+### Creating a Cluster Config
+
+```bash title="declaring a cluster config"
+DEPLOY_NAME=my-site # this should match one of your top-level directories in your deploy repo
+DNS_ZONE="zone.where.all.dns.entries.will.live" # all services will have DNS entries under here
+
+# assuming you are in the top-level of your deploy repo checkout
+mkdir -p management/manifests/argocd/
+
+cat << EOF > management/manifests/argocd/${DEPLOY_NAME}-cluster.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ${DEPLOY_NAME}-cluster
+  namespace: argocd
+  annotations:
+    dns_zone: "${DNS_ZONE}"
+    understack.rackspace.com/env: prod
+    understack.rackspace.com/role: site
+  labels:
+    argocd.argoproj.io/secret-type: cluster
+type: Opaque
+stringData:
+  name: "${DEPLOY_NAME}"
+  server: https://my-site-cluster:6443
+  config: |
+    # see link below for details
+
+EOF
+```
+
+This file is not complete yet because you must configure the access to the
+cluster by ArgoCD, please see [cluster config][argocd-decl-setup] for details
+on how to complete this.
+
+There are additional annotations supported by UnderStack as well for
+different configurations.
+
+`understack.rackspace.com/env`
+: Possible values are `prod` and `dev`. If the value is set to `dev` then
+  various settings can be overridden so allow an easier development experience.
+
+`understack.rackspace.com/role`
+: Possible values are `site`, `global`, and `aio`. Which defines what
+  services should be deployed to this cluster based on its role. The `aio`
+  role stands for "All In One" and combines both `site` and `global`.
+
+`dns_zone`
+: The DNS zone under which all services in this cluster will have their DNS
+  records created.
+
+`uc_repo_git_url`
+: URL to the UnderStack git repo to use. Can only be set for `dev` env
+  clusters.
+
+`uc_repo_ref`
+: Git reference to use of the UnderStack repo. Can only be set for `dev` env
+  clusters.
+
+`uc_deploy_git_url`
+: URL to the deploy git repo to use. Can only be set for `dev` env
+  clusters.
+
+`uc_deploy_ref`
+: Git reference to use of the deploy repo. Can only be set for `dev` env
+  clusters.
+
+`uc_skip_components`
+: string-ified JSON list of components or services from UnderStack to not
+  install in this cluster.
```

Comment:
This section would benefit from a real-world example, ideally covering all of the annotations. If not possible, at least one example for `uc_skip_components`

Reply:
Agreed. Can I take this as a follow on? I've already got a few follow ons I'm aiming to do.